source/link.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <queue>
+
+#include "packet.hpp"
+
+namespace sim {
+
+/**
+ * Unidirectional link from the source to a_next
+ */
+class Link {
+public:
+    Link(Device *a_next, int a_speed_mbps);
+
+    /**
+     * Update the source egress delay and schedule the arrival event
+     * based on the egress queueing and transmission delays.
+     */
+    void process(Packet a_packet);
+
```

Comment:
Maybe we should add

```suggestion
Device* get_next();
```
for the `Arrive` packet